File: algorithms/java/Graph/src/com/mahbub/algorithm/FloydWarshall_Demo.java
package com.mahbub.algorithm;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.util.Arrays;
import java.util.Scanner;

public class FloydWarshall_Demo {

    static int NODES, EDGES;
    static int[][] G;
    static int[][] p;

    static void floydWarshall() {
        for (int k = 0; k < NODES; k++) {
            for (int i = 0; i < NODES; i++) {
                for (int j = 0; j < NODES; j++) {
                    G[i][j] = G[i][k] + G[k][j];
                    p[i][j] = p[k][j];
                }
            }
        }
    }

    static void printPath(int i, int j) {
        if (i != j) printPath(i, p[i][j]);
        System.out.print(j + " ");
    }

    static void printMat() {
        for (int i = 0; i < NODES; i++) {
            for (int j = 0; j < NODES; j++) {
                System.out.print(G[i][j] + " ");
            }
            System.out.println();
        }
    }

    public static void main(String args[]) {
        Scanner scanner = inputFromFile();

        while (scanner.hasNext()) {
            NODES = scanner.nextInt();
            EDGES = scanner.nextInt();

            G = new int[NODES+1][NODES+1];
            p = new int[NODES+1][NODES+1];
            for (int i = 0; i <= NODES; i++) {
                Arrays.fill(G[i], Integer.MAX_VALUE);
            }
            for (int i = 0; i < EDGES; i++) {
                int u = scanner.nextInt();
                int v = scanner.nextInt();
                int c = scanner.nextInt();

                G[u][v] = G[v][u] = c;
            }

            floydWarshall();

            printMat();
        }
    }

    static Scanner inputFromFile() {
        try { return new Scanner(new FileInputStream("input.txt")); }
        catch (FileNotFoundException e) { e.printStackTrace(); }
        return null;
    }

    static Scanner inputFromSystem() {
        return new Scanner(System.in);
    }
}

File: algorithms/kotlin/Graph/FloydWarshall_Demo.kt
package com.mahbub.algorithm.graph

import java.io.FileInputStream
import java.io.FileNotFoundException
import java.util.*

class FloydWarshall_Demo {
    var NODES : Int = 0
    var EDGES : Int = 0

    var G : Array<IntArray> = emptyArray()
    var p : Array<IntArray> = emptyArray()

    fun floydWarshall() {
        for (k in 0 until NODES)
            for (i in 0 until NODES)
                for (j in 0 until NODES) {
                    G[i][j] = G[i][k] + G[k][j]
                    p[i][j] = p[k][j]
                }
    }

    fun printPath(i : Int, j : Int) {
        if (i != j) printPath(i, p[i][j])
        print("$j ")
    }

    fun printMat() {
        for (i in 0 until NODES)
            for (j in 0 until NODES)
                print("${G[i][j]} ")
        println()
    }

    fun fakeMain() {
        var scanner = inputFromFile()

        while (scanner!!.hasNext()) {
            NODES = scanner.nextInt()
            EDGES = scanner.nextInt()

            G = Array(NODES+1) { IntArray(NODES+1) {Int.MAX_VALUE} }
            p = Array(NODES+1) { IntArray(NODES+1) {Int.MAX_VALUE} }
            for (i in 0 until EDGES) {
                var u = scanner.nextInt()
                var v = scanner.nextInt()
                var c = scanner.nextInt()

                G[u][v] = c
                G[v][u] = c
            }

            floydWarshall()
            printMat()
        }
    }
}

fun main(args : Array<String>) {
    FloydWarshall_Demo().fakeMain()
}

private fun inputFromFile() : Scanner? {
    return try { Scanner(FileInputStream("input.txt")) }
    catch (e : FileNotFoundException) { null }
}

private fun inputFromSystem() : Scanner? {
    return Scanner(System.`in`)
}